### DISCUSSION TITLE
Not Serving HTTPS

### DISCUSSION THREAD

**Discussion by Agastya-Asthana** [coder/code-server#2595] [Q&A] [answered]
- Web Browser: Chromium 
- Local OS: Alpine Linux
- Remote OS: Raspbian/buster
- Remote Architecture: armv7
- `code-server --version`: 3.8.0

I am trying to dockerize the code-server for armv7 systems. When I run the `code-server` command in the container is outputs a bunch of logs and the last one is "[2020-12-24T03:49:41.745Z] info    - Not serving HTTPS." I have my docker container mapping port 8080 of host to the container's. The code-server doesn't work and I am pretty sure it's because of the error message above.

**Comment by fergy003** [comment]
I had the same issue this morning, which is how I ended up here. A guide I was following was missing this line in the docker compose file under # HTTP Routers
- "traefik.http.routers.vscode-rtr.tls=true"

Not sure if this will help or not, but it worked for me.

**Comment by Agastya-Asthana** [comment]
@fergy003 Could you share the guide you were using too?

**Comment by fergy003** [comment]
https://www.smarthomebeginner.com/wordpress-on-docker-traefik/

Before the main focus of this guide they mention vscode and the
docker-compose file

**Comment by nhooyr** [comment]
There's something else wrong with your setup. That line is expected.

**Comment by nooobcoder** [comment, marked as answer]
@Agastya-Asthana 
Here is a very good docker-run to help you with HTTP

STEP 1
Generate the self signed certificate file 
GENERATE CRT AND KEY FILE IN THE CERTS DIRECTORY
openssl req \
       -newkey rsa:2048 -nodes -keyout certs.key \
       -x509 -days 365 -out certs.crt

STEP 2 
Make a certs folder in your current working directory and then use below docker run
Replace _**your-username-here**_ in the below docker-run with your current user.

STEP 3
docker run -it --rm --name code-server-https -p 0.0.0.0:8080:8080   -v "$(pwd)/CodeServer/.config:/home/coder/.config"   -v "$(pwd)/VSCodes:/home/coder/project" -u "$(id -u):$(id -g)" -v $(pwd)/CodeServer/certs/certs.crt:/home/coder/cert.crt -v $(pwd)/CodeServer/certs/certs.key:/home/coder/cert.key -e "DOCKER_USER=your-username-here"  -e "USER=your-username-here"    codercom/code-server:latest --cert /home/coder/cert.crt --cert-key /home/coder/cert.key

STEP 4
All done, you should be good to go. Now bash in to the container and see that code-server is serving with HTTPS.

@Agastya-Asthana  **_### Please mark this discussion as answered if your query has been solved._**
Thank You
